### PR TITLE
Location sharing: Persist beacon info summaries to disk (PSF-1036)

### PR DESCRIPTION
**Pull request by SBiOSoftWhare**
https://github.com/vector-im/element-ios/issues/6199

### CODE REVIEW

File: MatrixSDK/Utils/Realm/RLMSupport.swift [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    public func objects(where predicateFormat: String, _ args: CVarArg...) -> RLMResults<NSObject> {
+        return objects(with: NSPredicate(format: predicateFormat, arguments: getVaList(args))) as! RLMResults<NSObject>
+    }
+}
+
+extension RLMCollection {
+    /// Allows for subscript support with RLMDictionary.
+    public subscript(_ key: String) -> AnyObject? {
+        get {
+            (self as! RLMDictionary<NSString, AnyObject>).object(forKey: key as NSString)
```
Comment (Johennes) [inline MatrixSDK/Utils/Realm/RLMSupport.swift:168]: Would `as?` be safer here given that the return value is an optional anyway?

Comment (SBiOSoftWhare) [reply MatrixSDK/Utils/Realm/RLMSupport.swift:168]: In fact this file come from Realm company. That's why I didn't touch it. But I agree with your comment.

Comment (Johennes) [reply MatrixSDK/Utils/Realm/RLMSupport.swift:168]: Ohh, sorry I missed the copyright comment at the top. 🤦‍♂️

All good then.

File: MatrixSDK/Aggregations/LocationSharing/Store/Realm/MXBeaconInfoSummaryRealmStore.swift [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+        }
+    }
+    
+    private func realmConfiguration(for userId: String) throws -> RLMRealmConfiguration {
+        let realmConfiguration = RLMRealmConfiguration.default()
+        
+        let rootDirectoryURL: URL
+        
+        do {
+            rootDirectoryURL = try FileManager.default.url(for: .cachesDirectory, in: .userDomainMask, appropriateFor: nil, create: true).appendingPathComponent(userId)
```
Comment (ismailgulek) [inline MatrixSDK/Aggregations/LocationSharing/Store/Realm/MXBeaconInfoSummaryRealmStore.swift:94]: We don't need to use shared container here?

**Review by ismailgulek — APPROVED** [review]
LGTM 👍

**Pull request merged**